app/customer.py

```diff
@@ -0,0 +1,28 @@
+from app.car import Car
+from decimal import Decimal
+import math
+
+
+class Customer():
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict[str, int],
+            location: list[int],
+            money: Decimal,
+            car: Car
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = car
+
+    def distance(self, shop_location: list) -> float:
+        return math.sqrt(
+            (self.location[0] - shop_location[0]) ** 2
+            + (self.location[1] - shop_location[1]) ** 2)
+
+    def ride_home(self, spend_money: float) -> None:
+        print(f"{self.name} rides home")
+        print(f"{self.name} now has {self.money - spend_money} dollars\n")

```

Comment:
In the `ride_home` method, the `money` attribute is not updated after spending money. You should update `self.money` by subtracting `spend_money` from it to reflect the current financial state of the customer.